tests/test_build_latex.py

```diff
@@ -1708,3 +1708,30 @@ def test_copy_images(app, status, warning):
         'rimg.png',
         'testimäge.png',
     }
+
+
+@pytest.mark.sphinx('latex', testroot='latex-labels-before-module')
+def test_duplicated_labels_before_module(app, status, warning):
+    app.build()
+    content: str = (app.outdir / 'python.tex').read_text()
+
+    def count_label(name):
+        text = r'\phantomsection\label{\detokenize{%s}}' % name
+        return content.count(text)
+
+    pattern = r'\\phantomsection\\label\{\\detokenize\{index:label-(?:auto-)?\d+[a-z]*}}'
+    expect_labels = {match.group() for match in re.finditer(pattern, content)}
+    result_labels = set()
```

Comment:
The naming is counterintuitive to me. I would rename `expect_labels` to `output_labels`(or even `result_labels`!) and `result_labels` to `input_labels`.  Or perhaps `labels_in_output` for the former and `labels_from_input` for the latter.

Reply:
Oh that's my bad. I totally swapped the two names ! To follow your advice I used `output_labels` and `tested_labels`.